libcodechecker/util.py

```diff
@@ -189,16 +189,23 @@ def split_server_url(url):
 
     LOG.debug("Parsing server url '{0}'".format(url))
 
-    protocol = 'http'
+    protocol = 'http'  # Default to using normal HTTP even if not written.
+    port = 8001  # Default port for CodeChecker.
     if url.startswith('http'):
         parts = url.split('://', 1)
         protocol = parts[0]
         url = url.replace(parts[0] + '://', '')
 
+        # Set port to standard ports if protocol was explicitly written.
+        if protocol == 'http':
```

Comment:
Create a helper function function for this because you use it in multiple places. Also use this newly created function in `unit/test_url_understanding.py` file.